appinventor/components/src/com/google/appinventor/components/runtime/util/ViewUtil.java

```diff
@@ -189,9 +196,39 @@ public static void setBackgroundImage(View view, Drawable drawable) {
   /**
    * Sets the image for an ImageView.
    */
-  public static void setImage(ImageView view, Drawable drawable) {
-    view.setImageDrawable(drawable);
-    if (drawable != null) {
+  public static void setImage(ComponentContainer container, ImageView view, String picturePath) {
+    GlideDrawableImageViewTarget imageViewTarget = new GlideDrawableImageViewTarget(view);
+    MediaUtil.MediaSource mediaSource = MediaUtil.determineMediaSource(container.$form(), picturePath);
+    switch (mediaSource) {
+      case ASSET:
```

Comment:
Why is the logic for assets different from any other? I would expect GIFs stored as assets should be playable just like any other. Could you explain this in a comment or reconcile everything to use the same approach?